### PR TITLE
Mmca 2240

### PR DESCRIPTION
**Pull request by kapil-sanghvi**

### CODE REVIEW

File: test/uk/gov/hmrc/customs/financials/emailthrottler/controllers/EmailThrottlerControllerSpec.scala [hunk 1]
```diff
@@ -44,17 +44,16 @@ class EmailThrottlerControllerSpec extends WordSpec with Matchers with MockitoSu
...
-    val controller = new EmailThrottlerController(mockEmailQueue, mockAppconfig, Helpers.stubControllerComponents())
+    val controller = new EmailThrottlerController(mockEmailQueue, Helpers.stubControllerComponents())
   }
 
   "the controller" should {
 
     "handle enqueue request" in new EmailThrottlerScenario {
       val result = controller.enqueueEmail()(fakeRequest)
-      status(result) shouldBe Status.ACCEPTED
+      status(result) mustBe Status.ACCEPTED
```
Comment (derekspalmer) [inline test/uk/gov/hmrc/customs/financials/emailthrottler/controllers/EmailThrottlerControllerSpec.scala:56]: :+1:

File: build.sbt [hunk 2]
```diff
@@ -1,18 +1,26 @@
...
+
 lazy val microservice = Project(appName, file("."))
   .enablePlugins(play.sbt.PlayScala, SbtAutoBuildPlugin, SbtGitVersioning, SbtDistributablesPlugin, SbtArtifactory)
   .settings(scoverageSettings: _*)
   .settings(
     majorVersion                     := 0,
-    libraryDependencies              ++= AppDependencies.compile ++ AppDependencies.test
+    libraryDependencies              ++= AppDependencies.compile ++ AppDependencies.test,
+    dependencyOverrides              ++= AppDependencies.overrides,
+    scalaVersion                     := "2.12.11",
```
Comment (derekspalmer) [inline build.sbt:17]: :+1:

File: project/AppDependencies.scala [hunk 3]
```diff
@@ -21,7 +20,7 @@ object AppDependencies {
   val akkaVersion = "2.5.23"
   val akkaHttpVersion = "10.0.15"
 
-  val overrides = Set(
+  val overrides: Seq[ModuleID] = Seq(
```
Comment (retroredge) [inline project/AppDependencies.scala:23]: Wondering why are these specific versions of the Akka libraries needed for this repo?

File: build.sbt (hunk 2 above)
[the remark this answers is not in the document]
Comment (kapil-sanghvi) [reply build.sbt:17]: Sorry, I don't understand what you mean there - I wanted it to be explicitly added there

File: build.sbt [hunk 4]
```diff
@@ -1,18 +1,26 @@
...
+organization := "uk.gov.hmrc"
+
 lazy val microservice = Project(appName, file("."))
   .enablePlugins(play.sbt.PlayScala, SbtAutoBuildPlugin, SbtGitVersioning, SbtDistributablesPlugin, SbtArtifactory)
   .settings(scoverageSettings: _*)
   .settings(
     majorVersion                     := 0,
-    libraryDependencies              ++= AppDependencies.compile ++ AppDependencies.test
+    libraryDependencies              ++= AppDependencies.compile ++ AppDependencies.test,
+    dependencyOverrides              ++= AppDependencies.overrides,
```
[the remark this answers is not in the document]
Comment (kapil-sanghvi) [reply build.sbt:16]: this was already there hanging somewhere in the build.sbt, so I have included that here like a closed knit config settings

File: project/AppDependencies.scala (hunk 3 above)
Comment (kapil-sanghvi) [reply project/AppDependencies.scala:23]: not sure, didn't try to look it - not in scope I think

File: build.sbt [hunk 5]
```diff
@@ -1,18 +1,26 @@
...
+    scalaVersion                     := "2.12.11",
+    scalacOptions                    := Seq("-target:jvm-1.8"),
```
[the remark this answers is not in the document]
Comment (kapil-sanghvi) [reply build.sbt:18]: yes, but I wanted to follow https://docs.scala-lang.org/overviews/compiler-options/index.html - which will help us further in adding any more compiler options

File: project/AppDependencies.scala (hunk 3 above)
Comment (kapil-sanghvi) [reply project/AppDependencies.scala:23]: It does fail If I remove those, so they are needed

File: build.sbt (hunk 5 above)
[the remark this answers is not in the document]
Comment (retroredge) [reply build.sbt:18]: OK are you going to retrofit this to the stubs then?

File: build.sbt (hunk 2 above)
[the remark this answers is not in the document]
Comment (kapil-sanghvi) [reply build.sbt:17]: I will add in stub as well

**Pull request merged**